packages/js/src/ui/components/primitives/Collapsible.tsx

```diff
@@ -1,32 +1,64 @@
-import { type Component, JSX, createEffect, createSignal } from 'solid-js';
+import { type Component, JSX, createEffect, createSignal, onCleanup } from 'solid-js';
 import { useStyle } from '../../helpers';
 
 type CollapsibleProps = JSX.IntrinsicElements['div'] & {
   class?: string;
   open: boolean;
 };
 
+const isInterpolateSizeSupported = () => {
+  return CSS.supports('interpolate-size', 'allow-keywords');
+};
+
 export const Collapsible: Component<CollapsibleProps> = (props) => {
+  const supportsInterpolateSize = isInterpolateSizeSupported();
   const style = useStyle();
   let contentRef: HTMLDivElement | undefined;
   const [enableTransition, setEnableTransition] = createSignal(false);
+  const [scrollHeight, setScrollHeight] = createSignal(0);
+
+  const updateScrollHeight = () => {
+    setScrollHeight(contentRef?.scrollHeight || 0);
+  };
 
   createEffect(() => {
     // Delay applying transitions until after the initial render
     requestAnimationFrame(() => setEnableTransition(true));
+
+    const resizeObserver = new ResizeObserver(() => {
+      updateScrollHeight();
+    });
+    if (contentRef && !supportsInterpolateSize) {
+      resizeObserver.observe(contentRef);
+    }
```

Comment:
use the resize observer as the fallback to the browsers that don't support `interpolate-size`